core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -2154,6 +2154,24 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
     (scalaTypeVal, SnappySession.getDataType(storeType, storePrecision, storeScale))
   }
+
+  /*
+     Method to add/update Structured Streaming UI Tab if cluster is running in embedded mode or
+     smart connector mode using SnappyData's Spark distribution
+  */
+  def updateStructuredStreamingUITab: Unit = {
+    try {
+      val updateUIMethod: Method = super.getClass.getMethod("updateUIWithStructuredStreamingTab")
+      updateUIMethod.invoke(this)
+    } catch {
+      case e: NoSuchMethodException =>
+        logWarning("Unable to add Structured Streaming UI Tab because" +
+            " updateUIWithStructuredStreamingTab method is not found in SparkSession class.")
```

Comment:
It will be good to indicate that in the message that this means snappy-spark distribution is not used.

Reply:
Yes, done. 
As discussed updated the log message to indicate it is not snappy-spark distribution if UI is unable to add structured streaming tab.